### ISSUE TITLE
Can composite builds span multiple different Eclipse workspaces?

### ISSUE THREAD

**Issue by ams-tschoening** [eclipse/buildship#791]
I have various libs in use by different applications and am using different workspaces for Eclipse for those libs and the apps. While the apps are checked out per workspace, the libs aren't, but managed in one workspace and then linked at project level to other workspaces. 

	Eclipse
		Workspace1
			lib1
			lib2
		Workspace2
			app1
			lib1 (link to Workspace1/lib1)
			lib2 (link to Workspace1/lib2)
		Workspace3
			app2
			lib1 (link to Workspace1/lib1)
			lib2 (link to Workspace1/lib2)

That approach seems to be a problem for Buildship in case those libs are part of a composite build of different apps, because it stores one reference to the including build in `org.eclipse.buildship.core.prefs` of the libs:

> connection.project.dir=../app1

In the concrete example, the containing lib is used in a workspace in which the configured path is not available, resulting in the following error message in Eclipse:

> Invalid Gradle project configuration file: .settings/org.eclipse.buildship.core.prefs

When I'm deleting the referenced path, the project compiles fine.

I guess Buildship is not able to handle my scenario and manage multiple including builds for those libs? What alternatives do I have?

The only thing coming into my mind is simply checking out each lib into the workspace it is needed in and don't use links to one and the same project anymore.

Thanks!

**Comment by donat** [comment]
You are correct, Buildship doesn't support your use-case at the moment. The problem is that the root project location is stored in the `.settings` the folder in all subprojects. The solution would be to move the project configuration to the workspace metadata.
